### PR TITLE
[tooling/46] Update Github Action versions

### PR DESCRIPTION
**Pull request by allanlewis**
In order to keep the `flex` workflows up-to-date, this PR updates the versions of the actions we use to the latest.

While working on this, I noticed that the integration tests were failing so I've made them pass and then updated them to take account of my substantive changes.

### CODE REVIEW

**Comment by allanlewis** [comment]
[Alert developers when any workflow fails on `main`](https://trello.com/c/fTrL90n7/46-alert-developers-when-any-workflow-fails-on-main)

File: tests/integration/tests.sh [hunk 1]
```diff
@@ -6,55 +6,45 @@
...
-    echo "$test_scenario seems to be WIP!"
+    echo "${test_scenario} seems to be WIP!"
     passed=1
     cd - > /dev/null
     continue
   else
     # Confirm that actual.yml matches expected.yml.
-    diff expected.yml $actual_filename > $diff_filename
-    if [[ $(wc -l $diff_filename | trim) == "0 $diff_filename" ]]
+    if diff -U3 expected.yml "${actual_filename}"
```
Comment (afshelley) [inline tests/integration/tests.sh:43]: According to `man diff`, `-U3` is the default. Are you just being verbose here, or do you expect it to change based on `diff` version or something?

**Review by afshelley — APPROVED** [review]
Looks good to me.

File: tests/integration/tests.sh (hunk 1 above)
Comment (allanlewis) [reply tests/integration/tests.sh:43]: [`man diff`][1] doesn't mention `-U3` - or do you mean that `3` is the default value for `-U`? In that case yes, I'm just being explicit. (The default is a "normal diff" which I find much harder to read.)

[1]: https://man7.org/linux/man-pages/man1/diff.1.html

**Pull request merged**